From 2922af379dc70b4b1a63b01d67179eb431f03ac4 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@redhat.com>
Date: Mon, 18 Mar 2024 11:14:54 -0700
Subject: [PATCH] REGRESSION(274077@main): failure to build on i586 (and likely
 other 32bit arches): static assertion failed: Timer should stay small
 https://bugs.webkit.org/show_bug.cgi?id=271108

Unreviewed build fix. This changes SameSizeOfTimer to ensure it matches
the size of Timer on 32-bit platforms.

* Source/WebCore/platform/Timer.cpp:

Canonical link: https://commits.webkit.org/276282@main

Upstream-Status: Backport [https://github.com/WebKit/WebKit/commit/2922af379dc70b4b1a63b01d67179eb431f03ac4]
---
 Source/WebCore/platform/Timer.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/Timer.cpp b/Source/WebCore/platform/Timer.cpp
index 4f7c0f5c39ca9..0f3734cca2474 100644
--- a/Source/WebCore/platform/Timer.cpp
+++ b/Source/WebCore/platform/Timer.cpp
@@ -263,7 +263,11 @@ struct SameSizeAsTimer {
 
     WeakPtr<TimerAlignment> timerAlignment;
     double times[2];
-    void* pointers[3];
+    void* pointers[2];
+#if CPU(ADDRESS32)
+    uint8_t bitfields;
+#endif
+    void* pointer;
 };
 
 static_assert(sizeof(Timer) == sizeof(SameSizeAsTimer), "Timer should stay small");
